content/intro-to-storybook/ember/en/composite-component.md

```diff
@@ -0,0 +1,301 @@
+  />
+</video>
+
+Note the position of the pinned item in the list. We want the pinned item to render at the top of the list to make it a priority for our users.
+
+## Automated testing
+
+Since `TaskList` adds another layer of complexity we want to verify that certain inputs produce certain outputs in a way amenable to automatic testing. To do this we’ll create unit tests using [Qunit](https://qunitjs.com/) coupled with a test renderer.
+
+![Qunit logo](/intro-to-storybook/logo-qunit.jpg)
+
+### Unit tests with Qunit
+
+Storybook stories paired with manual visual tests go a long way to avoiding UI bugs. If stories cover a wide variety of component use cases, and we use tools that ensure a human checks any change to the story, errors are much less likely.
+
+However, sometimes the devil is in the details. A test framework that is explicit about those details is needed. Which brings us to unit tests.
+
+In our case, we want our `TaskList` to render any pinned tasks **before** unpinned tasks that it has passed in the `tasks` prop. Although we have a story (`WithPinnedTasks`) to test this exact scenario, it can be ambiguous to a human reviewer that if the component **stops** ordering the tasks like this, it is a bug. It certainly won’t scream **“Wrong!”** to the casual eye.
+
+So, to avoid this problem, we can use Qunit to render the component and run some DOM querying code to verify salient features of the output.
+
+Create a test file called `tests/integration/task-list-test.js`. Here, we’ll build out our tests that make assertions about the output.
+
+```javascript
+// tests/integration/task-list-test.js
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+module('Integration | Component | TaskList', function(hooks) {
+  setupRenderingTest(hooks);
+  const taskData = {
+    id: '1',
+    title: 'Test Task',
+    state: 'TASK_INBOX',
+    updatedAt: new Date(2018, 0, 1, 9, 0),
+  };
+  const tasklist = [
+    { ...taskData, id: '1', title: 'Task 1' },
+    { ...taskData, id: '2', title: 'Task 2' },
+    { ...taskData, id: '3', title: 'Task 3' },
+    { ...taskData, id: '4', title: 'Task 4' },
+    { ...taskData, id: '5', title: 'Task 5' },
+    { ...taskData, id: '6', title: 'Task 6 (pinned)', state: 'TASK_PINNED' },
+  ];
+
+  test('renders pinned tasks at the start of the list', async function(assert) {
+    this.set('tasks', tasklist);
+    this.set('testpinTask', actual => {
+      let expected = 1;
+      assert.deepEquals(actual, expected);
+    });
+
+    this.set('testarchiveTask', actual => {
+      let expected = 1;
+      assert.deepEquals(actual, expected);
+    });
+
+    await render(
+      hbs`{{TaskList tasks=tasks pinTask=(action testpinTask) archiveTask=(action testarchiveTask) }}`
+    );
+    assert.dom(this.element.querySelector('.list-item:nth-of-type(1)')).hasClass('TASK_PINNED');
```

Comment:
```suggestion
    assert.dom('.list-item:nth-of-type(1)').hasClass('TASK_PINNED');
```

Usually, you would add test-selectors, e.g. `[data-test-*]` to your html elements and use them in tests, as this would make it independent from used CSS classes, much safer!